docs/sdks/python/knowledge-bases.mdx

```diff
@@ -0,0 +1,1069 @@
+    +-----+----------------------+------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+-------------------+
+    | A1B | A1B_notes:1of1:0to20 | Request color: black   | {"chunk_index":0,"content_column":"notes","end_char":20,"original_doc_id":"A1B_notes","original_row_id":"A1B","product":"Wireless Mouse","source":"TextChunkingPreprocessor","start_char":0} | 0.5743341242061104 | 0.504396172197583 |
+    +-----+----------------------+------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------+-------------------+
+    ```
+  </Step>
+</Steps>
+
+The following sections explain the syntax and other features of knowledge bases.
+
+## `create()` Function
+
+Here is the syntax for creating a knowledge base:
+
+```python
+my_kb = project.knowledge_bases.create(
+    'my_kb',
+    embedding_model={
+        'provider': 'openai', 
+        'model_name': 'text-embedding-3-small', 
+        'api_key': 'sk-...'},
+    reranking_model={
+        'provider': 'openai', 
+        'model_name': 'gpt-4', 
+        'api_key': 'sk-...'},
+    storage=server.databases.my_db.tables.my_table,
+    metadata_columns=['date', 'creator', ...],
+    content_columns=['review', 'content', ...],
+    id_column='id'
+)
+```
+
+Upon execution, it registers `my_kb` and associates the specified models and storage. `my_kb` is a unique identifier of the knowledge base within MindsDB.
+
+<Note>
+As MindsDB stores objects, such as models or knowledge bases, inside [projects](/mindsdb_sql/sql/create/project), you can create a knowledge base inside a custom project.
+
+```python
+project = server.create_project('my_project')
+
+my_kb = project.knowledge_bases.create(...)
+```
+</Note>
+
+### Supported LLMs
+
+Below is the list of all language models supported for the `embedding_model` and `reranking_model` parameters.
+
+#### `provider = 'openai'`
+
+When choosing `openai` as the model provider, users should define the following model parameters.
+
+* `model_name` stores the name of the OpenAI model to be used.
+* `api_key` stores the OpenAI API key.
+
+Learn more about the [OpenAI integration with MindsDB here](/integrations/ai-engines/openai).
+
+#### `provider = 'openai_azure'`
+
+When choosing `openai_azure` as the model provider, users should define the following model parameters.
+
+* `model_name` stores the name of the OpenAI model to be used.
+* `api_key` stores the OpenAI API key.
```

Comment:
Did we change the keys without the provider name? just to double check or this should be `openai_api_key`?